test/distributed/checkpoint/test_state_dict.py

```diff
@@ -355,3 +355,18 @@ def init_model_optim():
             return orig_model, orig_optim, copy_optim, model_copy, optim_copy
 
         self._test_save_load(init_model_optim)
+
+    @skip_if_lt_x_gpu(1)
+    def test_strict(self) -> None:
+        model = CompositeParamModel(device=torch.device("cuda"))
+
+        model_state_dict, _ = state_dict(model)
+        key = next(iter(model_state_dict.keys()))
+        model_state_dict.pop(key)
+        with self.assertRaisesRegex(RuntimeError, "Missing key"):
+            load_state_dict(model, model_state_dict=model_state_dict)
+        load_state_dict(
```

Comment:
I believe the nn.module.load_state_dict will return `missing_keys, unexpected_keys` tuple, will this API do the same?

Reply:
No, it will not. We can add the support.

Reply:
I'll add the return values in a separate PR.